src/Rules/Operators/InvalidBinaryOperationRule.php

```diff
@@ -127,7 +128,10 @@ public function processNode(Node $node, Scope $scope): array
 				&& !$this->hasDivisionByZeroThrowsTag($scope)
 			) {
 				$zeroType = new ConstantIntegerType(0);
-				if ($zeroType->isSuperTypeOf($rightType)->maybe()) {
+				if (
+					$zeroType->isSuperTypeOf($rightType)->maybe()
+					&& !$rightType instanceof BenevolentUnionType
```

Comment:
This wouldn't go against `checkBenevolentUnionTypes` ?
https://phpstan.org/config-reference#checkbenevolentuniontypes

Shouldn't it be
```
&& ($this->checkBenevolentUnionTypes || !$rightType instanceof BenevolentUnionType)
```

Reply:
I think the flag is working, because we are using the RuleLevelHelper, see https://github.com/phpstan/phpstan-src/pull/3419/files#diff-cb8642718212e42a4ffea566a1b6d116db66af1685c26e4effdba3ce72539302R883-R886